operator/mapper/postgresql.go

```diff
@@ -0,0 +1,135 @@
+package mapper
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/exoscale/egoscale/v2/oapi"
+	"github.com/minio/minio-go/v7/pkg/set"
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+	"k8s.io/utils/pointer"
+)
+
+// PostgreSQLMapper is a mapper to convert exoscale's SDK resources to Kubernetes API specs.
+type PostgreSQLMapper struct{}
```

Comment:
Any reason to create an empty struct here? Any reason to not use simple functions for conversion?

Reply:
Mainly Naming :tm: I expect that there will be other mappers in the future, e.g. Mysql, Kafka etc. So I didn't want to end up with long function names like `ToPostgreSQLSpec(...)` or `ToPostgreSQLConnectionDetails(...)`.